app/models/health/ssm_saver.rb

```diff
@@ -23,23 +22,29 @@ def create
     def update
       @ssm.class.transaction do
         if @ssm.completed_at.present? && @ssm.completed_at_changed? && @create_qa
-          @qualifying_activity = Health::QualifyingActivity.where(
-            source_id: @ssm.id,
-            source_type: @ssm.class.name,
-          ).first_or_initialize
-          @qualifying_activity.assign_attributes(
-            user_id: @user.id,
-            user_full_name: @user.name_with_email,
-            date_of_activity: @ssm.completed_at,
-            activity: :cha,
-            follow_up: 'Improve Patient Outcomes',
-            reached_client: :yes,
-            mode_of_contact: :in_person,
-            patient_id: @ssm.patient_id
-          )
+          @cha = @ssm.patient.recent_cha_form
+          if @cha.completed? && @cha.reviewed?
+            qualifying_activity = setup_qualifying_activity
+            qualifying_activity.save
```

Comment:
This appears to be the inverse.  So maybe the comment is "Only add a QA after the CHA _and_ SSM have been completed.

Reply:
Good idea.